src/Domain/Entity/ContactMediaMap.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace BBC\ProgrammesPagesService\Domain\Entity;
+
+use BBC\ProgrammesPagesService\Domain\Enumeration\ContactMediaEnum;
+use BBC\ProgrammesPagesService\Domain\ValueObject\ContactMedia;
+use InvalidArgumentException;
+
+class ContactMediaMap
+{
+    /**
+     * Example:
+     *  [
+     *      'email'    => [<ContactMedia>, <ContactMedia>, <ContactMedia>],
+     *      'facebook' => [<ContactMedia>],
+     *  ]
+     *
+     * @var ContactMedia[][]
+     */
+    private $contactsMapByMedia = [];
+
+    /**
+     * Example: ['reddit', 'email']
+     *
+     * @return string[]
+     */
+    public function getNamesOfUsedMedia(): array
+    {
+        return array_keys($this->contactsMapByMedia);
+    }
+
+    /**
+     * @return ContactMedia[]
+     */
+    public function getContactsByMedia(string $contactMediaType): array
+    {
+        $validsMedia = ContactMediaEnum::VALID_MEDIA;
+
+        if (!isset($validsMedia[$contactMediaType])) {
+            throw new InvalidArgumentException('Trying to get a contact detail with unknown type');
```

Comment:
I don't think we should throw an error here and instead power through, not adding an invalid thing to the list.

We don't want a page to start throwing 500 errors because some invalid thing snuck its way into isite.